policy/scan/local_scanner_test.go

```diff
@@ -100,3 +105,224 @@ func TestCreateAssetList(t *testing.T) {
+		Schema:        s.schema,
+		RemoveFailing: true,
+	})
+	s.Require().NoError(err)
+
+	s.job.Bundle = bundle
+	s.job.PolicyFilters = []string{"asset-policy"}
+	bundleMap := bundle.ToMap()
+
+	ctx := context.Background()
+	scanner := NewLocalScanner(WithRecording(providers.NullRecording{}))
+	res, err := scanner.RunIncognito(ctx, s.job)
+	s.Require().NoError(err)
+	s.Require().NotNil(res)
+
+	full := res.GetFull()
+	s.Require().NotNil(full)
+
+	s.Equal(1, len(full.Reports))
+
+	for k, r := range full.Reports {
+		// Verify the score is 100
+		s.Equal(uint32(100), r.GetScore().Value)
+
+		p := full.ResolvedPolicies[k]
+
+		// Get the code id for all the executed queries
+		executedQueries := []string{}
+		for qCodeId := range p.ExecutionJob.Queries {
+			executedQueries = append(executedQueries, qCodeId)
+		}
+
+		expectedQueries := []string{
+			bundleMap.Queries["//local.cnspec.io/run/local-execution/queries/ignored-query"].CodeId,
+			bundleMap.Queries["//local.cnspec.io/run/local-execution/queries/sshd-score-01"].CodeId,
+		}
+		s.ElementsMatch(expectedQueries, executedQueries)
+
+		queryIdToReportingJob := map[string]*policy.ReportingJob{}
+		for _, rj := range p.CollectorJob.ReportingJobs {
+			_, ok := queryIdToReportingJob[rj.QrId]
+			s.Require().False(ok)
+			queryIdToReportingJob[rj.QrId] = rj
+		}
+
+		// Make sure the ignored query is ignored
+		queryRj := queryIdToReportingJob[bundleMap.Queries["//local.cnspec.io/run/local-execution/queries/ignored-query"].CodeId]
+		s.Require().NotNil(queryRj)
+
+		parent := queryRj.Notify[0]
+		parentJob := p.CollectorJob.ReportingJobs[parent]
+		s.Require().NotNil(parentJob)
+		s.Equal(explorer.ScoringSystem_IGNORE_SCORE, parentJob.ChildJobs[queryRj.Uuid].Scoring)
+	}
+}
+
+func (s *LocalScannerSuite) TestRunIncognito_QueryExceptions() {
+	bundle, err := policy.BundleFromPaths("./testdata/exceptions.mql.yaml")
+	s.Require().NoError(err)
+
+	_, err = bundle.CompileExt(context.Background(), policy.BundleCompileConf{
+		Schema:        s.schema,
+		RemoveFailing: true,
+	})
+	s.Require().NoError(err)
+
+	s.job.Bundle = bundle
+	s.job.PolicyFilters = []string{"asset-policy"}
+	bundleMap := bundle.ToMap()
+
+	ctx := context.Background()
+	scanner := NewLocalScanner(WithRecording(providers.NullRecording{}))
+	res, err := scanner.RunIncognito(ctx, s.job)
+	s.Require().NoError(err)
+	s.Require().NotNil(res)
+
+	full := res.GetFull()
+	s.Require().NotNil(full)
+
+	s.Equal(1, len(full.Reports))
+
+	for k, r := range full.Reports {
+		// Verify the score is 100
+		s.Equal(uint32(100), r.GetScore().Value)
+
+		p := full.ResolvedPolicies[k]
+
+		// Get the code id for all the executed queries
+		executedQueries := []string{}
+		for qCodeId := range p.ExecutionJob.Queries {
+			executedQueries = append(executedQueries, qCodeId)
+		}
+
+		expectedQueries := []string{
+			bundleMap.Queries["//local.cnspec.io/run/local-execution/queries/ignored-query"].CodeId,
```

Comment:
I think, I'm missing something here.

Regarding the comment above about "IGNORE"/"DISABLE":
"ignored query" as an unspecified action, but is part of an ignored group. With the above code I'd not expected it to be included.

Reply:
yeah, the resolver code is complex. That's why I added tests for the old exception scenarios and the new ones. If you revert my changes to the resolver, the old test cases still pass. So that is the expected behaviour

Reply:
Thanks for the explanation. Makes sense to me.